apps/workers/email-composer/src/emailComposer.ts

```diff
@@ -0,0 +1,54 @@
+import { Message } from 'amqplib'
+import { EmailQueues } from 'shared/src/enums/emailQueues'
+import { createRabbitMqChannel } from 'shared/src/queue/createRabbitMqChannel'
+import { CONFIG } from '../config'
+import { ConsumeMessageReturn, consumeMessage } from './consumeMessage'
+import { getHtmlRoles } from './getHtmlRoles'
+
+const rabbitMqCredentials = {
+  password: CONFIG.RABBITMQ_PASS,
+  user: CONFIG.RABBITMQ_USER,
+}
+
+export const emailComposer = async () => {
+  const emailPreRendererChannel = await createRabbitMqChannel(
+    rabbitMqCredentials
+  )
+  const emailComposerChannel = await createRabbitMqChannel(rabbitMqCredentials)
+
+  await Promise.all([
+    emailPreRendererChannel.assertQueue(EmailQueues.EmailPreRenderer),
+    emailComposerChannel.assertQueue(EmailQueues.EmailComposer, {
+      durable: false,
+    }),
+  ])
+
+  const sendToEmailComposerQueue = async (props: Record<string, string>) => {
+    emailComposerChannel.sendToQueue(
+      EmailQueues.EmailComposer,
+      Buffer.from(JSON.stringify(props))
+    )
+  }
+
+  let emailPreRendererMessage = await emailPreRendererChannel.get(
+    EmailQueues.EmailPreRenderer
+  )
+  do {
+    const sanitizedMessage = await consumeMessage(emailPreRendererMessage)
+    if (!sanitizedMessage) break
+    const { email, footerHTML, headerHTML, roles } =
+      sanitizedMessage as ConsumeMessageReturn
+    const rolesHTML = await getHtmlRoles(roles)
+
+    sendToEmailComposerQueue({
+      [email]: `${headerHTML}${rolesHTML}${footerHTML}`,
+    })
+    emailPreRendererChannel.ack(emailPreRendererMessage as Message)
+
+    emailPreRendererMessage = await emailPreRendererChannel.get(
+      EmailQueues.EmailPreRenderer
+    )
```

Comment:
This is wrong, we should not be getting one by one from the queue.

You need to add a callback to the consume method of the queue.

Reply:
Solved?
https://github.com/ocodista/trampar-de-casa/blob/2e50ce39e9f29ee7a9c26d4ba357297b8e6a40fe/apps/workers/email-composer/src/emailComposer.ts#L30-L34